### PR TITLE
fix problematic links and modifications

### PR DESCRIPTION
**Pull request by GaribMurshudov**
I have changed some links and modifcations: MET-TYR, SF32-CYS --> SF37-CYS, FOR-LYZ. They need to be tested. Link instruction for MET-TYR
LINK: RES-NAME-1 MET ATOM-NAME-1 SD RES-NAME-2 TYR ATOM-NAME-2 CE1 CHANGE CHARGE 1 SD 1

### CODE REVIEW

File: list/mon_lib_list.cif [hunk 1]
```diff
@@ -42103,6 +42119,69 @@ _chem_mod_tor.new_value_angle
...
+_chem_mod_tor.function
+_chem_mod_tor.atom_id_1
+_chem_mod_tor.atom_id_2
+_chem_mod_tor.atom_id_3
+_chem_mod_tor.atom_id_4
+_chem_mod_tor.id
+_chem_mod_tor.new_value_angle
+_chem_mod_tor.new_value_angle_esd
+_chem_mod_tor.new_period
+LYZm1          delete         CD        CE        NZ        HZ3       .           .           .           .
```
Comment (keitaroyam) [inline list/mon_lib_list.cif]: Current gemmi (0.6.0) does not allow "." for _chem_mod_tor.new_period, that's why the test failed. Upcoming 0.6.1 will be ok with it.

Comment (GaribMurshudov) [reply list/mon_lib_list.cif]: Ok. I see. I need to use new acedrg.

File: list/mon_lib_list.cif [hunk 2]
```diff
@@ -36103,7 +36103,7 @@ LYS-CYS LYS LYSmod1 peptide CYS CYSmod1 peptide LYS-CYS
 LYS-RET LYS LYSmod2 peptide RET RETmod1 NON-POLYMER LYS-RET
 LYS-ASN LYS LYSmod3 peptide ASN ASNmod1 peptide LYS-ASN
 MDO-pept MDO MDOmod1 NON-POLYMER . ALAmod1 peptide MDO-pept
-MET-TYR MET METmod1 peptide TYR TYRmod3 peptide MET-TYR
+MET-TYR MET METm1   peptide TYR TYRm3   peptide MET-TYR
```
Comment (keitaroyam) [inline list/mon_lib_list.cif:36106]: TYRm3 is not defined below

**Comment by keitaroyam** [comment]
I think it's better to keep the name TYRmod3 (rename TYRm1), and wrong references (see above) should be also fixed.

**Comment by GaribMurshudov** [comment]
Do you think METmod1 also should stay. Current version of acedrg makes TYRm3, METm1 etc. 
That is the problem with manual addition. There are always errors
I have changed TYRm3 to TYRmod3 and tried to fix related issues. 
We need automatic update. Otherwise the problems will accummulate.
We also need to start thinking about atom type dependent links. Now acedrg writes out acedrg atom types. We need to think how to use them. There should be a seperate topic on this matter.

**Comment by keitaroyam** [comment]
I am working on it but do not think it will be possible so soon. By the way do you have any example PDB for these updated links?

**Comment by GaribMurshudov** [comment]
> I am working on it but do not think it will be possible so soon. By the way do you have any example PDB for these updated links?

Examples where in the list of all possible links (essentially close contacts) calculate by Rob. I will try to find them.
It is another issue: we need an example pdb files (at least one, several would be good, if we would have all pdb files it would be even better) for each link. They should stored in the same place (or close by) as the list of link instructions.

**Comment by keitaroyam** [comment]
OK, maybe we can find all examples after fixing all problematic links.

**Comment by GaribMurshudov** [comment]
> OK, maybe we can find all examples after fixing all problematic links.

Let us do this. 
I need an automatic update of mon_lib_list.cif. Otherwise every time I add something there will be a new error.

**Comment by keitaroyam** [comment]
I fixed links related to 4D4 and MS6, as I needed them. I deleted pept-4D4 and 4D4-pept since they can be handled by the alias mechanism.

**Comment by GaribMurshudov** [comment]
It seems that the number of errors getting smaller.

**Comment by keitaroyam** [comment]
Should we keep NH1/NH2/NH3 modifications?

**Comment by GaribMurshudov** [comment]
They are not needed anymore (I think). But after removing them we should test refmac. If they are used they would be used in refmac. I hope I removed all references to these modifications. We also do not need COO (I think)

**Comment by keitaroyam** [comment]
OK, we should come back later. I updated following links to make test_monlib.TestMonlib happy. mod_DELk-O2 and mod_DEL-O1 will be revisited after sorting out sugars.

02J-pept
CYS-CYC
HIS_TYR1
DEL-NMH (mod)
PJE-010
CYS-PEB
NH2 (mod, just fixed id)

It seems several planes were unnecessarily removed in the original modifications (CYCmod1, PEBmod1, TYRmod1).

I used Acedrg 271 and following parts were fixed manually:
```
MS6-pept sp2_sp2_1 1 S 1 C 2 N 2 CA 0.000 5.0 2
```
changed to CA-CA, 180

```
HIS_TYR1 sp2_sp3_2 1 CB 1 CG 1 ND1 2 CB 0.000 0.0 2
```
changed to const*, period 1

**Comment by keitaroyam** [comment]
I updated following links together with TRP/TYR/GTP monomers to avoid redundant modifications.
CR8-pept GTP-p HIS_TYR2 TRP-TYR pept-CR8

Following links need to be updated, and I will wait for acedrg problems to be fixed.
CYS-FAD HIS-FAD1 HIS-FAD2

This branch should be carefully tested before merging to master.

**Comment by keitaroyam** [comment]
Now all errors in CI are sorted. But there may be an error in chemistry. Later will find example pdbs and test refinement to merge this to master.

**Comment by GaribMurshudov** [comment]
For some reason I cannot approve. Somebody else wil need to approve.
We will need to think about chemistry a bit more.
current thought (needs to be clarified, discssed and implemented)
1) count bond orders and make sure that they are consistent
2) check ligands in their nattive environment (in macromolecules). This will take soem time
3) Pattern match local substractures to those from COD

**Comment by keitaroyam** [comment]
This is because you started this PR. I can approve, but it's too early to do it. Let us do tests for updated links.

**Comment by keitaroyam** [comment]
It seems FOR-LYZ link does not exist in PDB. Why was it added? Perhaps FOR-LYS was intended?

**Comment by GaribMurshudov** [comment]
IT could well be. Is there FOR-LYS?

**Comment by keitaroyam** [comment]
Yes, but only three examples: 1pw1 2rus 3tax
(note this is just from LINK records; more links may be found if we add FOR-LYS)

**Comment by GaribMurshudov** [comment]
Ok. Let us replace FOR-LYZ with FOR-LYS

**Comment by keitaroyam** [comment]
OK, but this kind of "group" monomers (FOR is a formyl group) are used in several links (and sometimes it bridges two monomers). We would need to define many links in the current mechanism.

**Comment by GaribMurshudov** [comment]
I do not think we should increase the number of links at the moment. Instead we should work on new mechanisms for links. Now, acedrg writes out acedrg atom types. I was thinking to use these atom types for link definitions. At the moment the idea is not fully ready yet.

**Comment by keitaroyam** [comment]
Done. I think we can also delete AHT-ALA, as AHT is no longer used. FOR_C-C does not also exist in PDB and perhaps we can remove it, while this may be added for completeness when FOR_C-N was added.

**Comment by GaribMurshudov** [comment]
Yes. We should remove any link that does not exist in the pdb or is impossible. 
What was AHT?

**Comment by keitaroyam** [comment]
It seems AHT was used in 1tmb, and probably replaced by VLT.

**Comment by GaribMurshudov** [comment]
I see. We even do not have AHT. I agree that we should remove all links referenced to non-existing ligands

**Comment by keitaroyam** [comment]
We have AHT, but may not need it.

I tested updated links with high resolution PDBs. It seems the most problematic one is SF37-CYS (in 6fpo, 5adu etc). SF3 should also be fixed probably.

Others are more or less ok. A concern is planarity in bridged planes, as in MET-TYR, TRP-TYR, and HIS_TYR2 (in 5jhx and 7coh).

Planarity issue also happens in a monomer: 2gw3 B/CR8 63/C17 and 2oln A/FAD 1001/C8M for example.

Another concern may be sp2_sp2 sigmas, and we may come back later? For example PJE-010's sp2_sp2_1 as in 2q6f and in CYC and PEB.

**Comment by keitaroyam** [comment]
Updated a test, and found new problems (DEL-OXT not applicable to SUI, ORN-NE removing undefined HE3). Fixed, but not tested yet.

**Comment by keitaroyam** [comment]
pept-ORN/pept-SUI/SUI-pept all seem to be fine. Fe-S clusters should be handled in another issue.

**Review by keitaroyam — APPROVED** [review]

**Comment by GaribMurshudov** [comment]
Ok. Let us open another issue

**Pull request merged**